src/shogun/mathematics/graph/CMakeLists.txt

```diff
@@ -71,70 +71,73 @@ set_target_properties(
   LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/src/shogun)
 set_property(TARGET shogun-engine APPEND PROPERTY COMPATIBLE_INTERFACE_STRING ${SHOGUN_VERSION_MAJOR})
 
-add_library(shogun-vanilla-executor SHARED
-	${SHOGUN_GRAPH_IMPLEMENTATION_HEADERS}
-	${SHOGUN_GRAPH_IMPLEMENTATION_SOURCES}
-)
-target_link_libraries(shogun-vanilla-executor shogun-engine)
-# todo create a macro/function for this
-set_property(TARGET shogun-vanilla-executor PROPERTY POSITION_INDEPENDENT_CODE TRUE)
-set_target_properties(
-  shogun-vanilla-executor PROPERTIES
-  VERSION ${LIBSHOGUNVER}
-  SOVERSION ${LIBSHOGUNSO}
-  INTERFACE_shogun_MAJOR_VERSION ${SHOGUN_VERSION_MAJOR}
-  LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/src/shogun)
-set_property(TARGET shogun-vanilla-executor APPEND PROPERTY COMPATIBLE_INTERFACE_STRING ${SHOGUN_VERSION_MAJOR})
-LIST(APPEND EXCUTORS shogun-vanilla-executor)
+set(${EXECUTORS} "")
+
+function(add_executor EXECUTOR_NAME)
+
+    cmake_parse_arguments(
+        EXECUTOR
+        ""
+        "ADDITIONAL_SHARED_LIBRARIES;ADDITIONAL_INCLUDE_DIRS"
+        "IMPLEMENTATION_HEADERS;IMPLEMENTATION_SOURCES"
+        ${ARGN}
+    )
```

Comment:
@vigsterkr no idea if this how cmake_parse_arguments is meant to be used, but it seems to work